integr8ly/integreatly-operator
# INTLY-3288 - Add UPS Install

## Description
Using the Unified Push Server (UPS) Operator, install UPS as part of the integreatly install

The associated pull request for adding the UPS cluster service version:
* https://github.com/integr8ly/manifests/pull/70

## Verification
* Create a operator source that points to my app registry:
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: kevfan
  type: appregistry
```
* Browse to the operator hub and verifiy that a Unifiedpush Operator with version `0.2.0` is present
* Install Integreatly using the following Installation.cr as an example point to your cluster:
```
apiVersion: integreatly.org/v1alpha1
kind: Installation
metadata:
  name: example-installation
spec:
  type: workshop
  namespacePrefix: integreatly-
  routingSubdomain: apps.cluster-id.openshiftworkshop.com
  masterUrl: https://console-openshift-console.apps.cluster-id.openshiftworkshop.com
  apiUrl: https://api.cluster-id.openshiftworkshop.com
  selfSignedCerts: true
```
* Verify that the `integreatly-ups` namespace is created
* Verify that the `unifiedpush-operator` pod is created
* Verify that the `ups` custom resource install is created and resources are porvisioned correctly
* Verifiy that the installation custom resource yaml correctly updated the phase and exposed UPS route
* Visit exposed ups route to ensure it works
  * Note: You may get a internal 500 error when trying to login due to https://github.com/aerogear/unifiedpush-operator/pull/75 and the manifest may not be updated to use a image tag with this change (can be resolved manually by updating oauth-proxy image in image stream to `quay.io/openshift/origin-oauth-proxy:4.2.0` )
